### DISCUSSION TITLE
How to open and run the converted kotlin code in Android Studio?

### DISCUSSION THREAD

**Discussion by Yinzhi0409** [skiptools/skip#280] [Q&A] [answered]
Hi developers,
It's so glad and fantasitic to find such a convenient app to convert my ios project to aos project! Love your work.
However, I know I can check relative code in /project-name/Android, and I want to add some Android-specific components in the aos verision app.
Therefore I cpoy a Android folder file and open it in Android Studio. But I find the direction structure is different from standard andriod project. I even can't search the text in some buttons. It seems I can't just change the UI text in Android Studio. It seems to need to change the respective text in the ios project.
To summary, my qustions are:
1.Can I open the generated android project and run it in Android Studio? If can, how can I do that?
2.Can I just change the generated aos project in Android Studio? If can, how can I do that?
Thanks answers a lot! Look forward replies.

**Comment by marcprux** [comment, marked as answer]
You can open and run your app in Android Studio: https://skip.tools/docs/platformcustomization/#android-studio

This is useful for seeing the generated Kotlin and running the app in a debugger. However, you cannot change the generated Kotlin, since the Swift to Kotlin transpiler only goes one way. However, you can integrate raw Kotlin with your Skip app, which you *can* edit in Android Studio: https://skip.tools/docs/platformcustomization/#kotlin-files

**Reply by Yinzhi0409** [reply]
Thanks for your clear reply and instruction!
